tools/justfile

```diff
@@ -0,0 +1,156 @@
+# Default values for the deployment
+tag    := "test"
+queue  := "active"
+worker := "htcmock"
+
+# List recipes and their usage
+@default:
+  just --list
+  just _usage
+
+_usage:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  echo
+  echo "The recipe deploy takes three arguments"
+  echo
+  echo "usage: just tag=<tag> queue=<queue> worker=<worker> deploy"
+  echo "       if any of the variables is not set, its default value is used"
+  echo
+  echo "   tag: The core tag image to use, defaults to test"
+  echo
+  echo "   queue: allowed values below"
+  cat <<-EOF
+        active   :  for activemq (1.0.0 protocol) (default)
+        rabbit   :  for rabbitmq (1.0.0 protocol)
+        rabbit091:  for rabbitmq (0.9.1 protocol)
+        artemis  :  for artemismq (1.0.0 protocol)
+  EOF
+  echo
+  echo "   worker: allowed values below"
+  cat <<-EOF
+        htcmock: for HtcMock V3 (default)
+        stream: for Stream worker
+  EOF
+
+# Sets the queue
+_set-queue:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  
+  # Save variables in .env file to pass them to other recepies
+  echo "" > justcompose.env
+
+  OVERRIDE="../docker-compose/docker-compose.override.yml"
+  if [ "x{{ queue }}" == "xactive" ]
+  then
+    QUEUE="../docker-compose/docker-compose.queue-activemqp.yml"
+  elif [ "x{{ queue }}" == "xrabbit" ]
+  then
+    QUEUE="../docker-compose/docker-compose.queue-rabbitmq.yml"
+  elif [ "x{{ queue }}" == "xrabbit091" ]
+  then
+    QUEUE="../docker-compose/docker-compose.queue-rabbitmq.yml"
+    OVERRIDE="../docker-compose/docker-compose.override-rabbitmq091.yml"
+  elif [ "x{{ queue }}" == "xartemis" ]
+  then
+    QUEUE="../docker-compose/docker-compose.queue-artemis.yml"
+  else
+    echo "{{ queue }} is not a valid queue"
+    exit 1
+  fi
+  echo "export QUEUE=$QUEUE" >> justcompose.env
+  echo "export OVERRIDE=$OVERRIDE" >> justcompose.env
+
+# Defines worker and enviroment variables for deployment
+_set-vars:
```

Comment:
This should be done in `just` variables, and skip the `justcompose.env` file.

Reply:
I agree with you, but just doesn't support this: https://github.com/casey/just#setting-variables-in-a-recipe

Reply:
You don't need a recipe, you can use a if-then-else outside a rule, at the place you define the variable.

Reply:
Ah right!  that will be much nicer! I'm on it